src/inference/include/openvino/runtime/infer_request.hpp

```diff
@@ -315,6 +315,12 @@ class OPENVINO_RUNTIME_API InferRequest {
      */
     std::vector<VariableState> query_state();
 
+    /**
+     * @brief Resets all internal variable states for relevant infer request to a value specified as
+     * default for the corresponding `ReadValue` node
+     */
+    void reset_state();
```

Comment:
let's also reuse this function here
https://github.com/openvinotoolkit/openvino/blob/b31aad78b0c509fcc3583ff8f7d6ac2968bdfc60/src/tests/functional/shared_test_classes/include/shared_test_classes/subgraph/stateful_model.hpp#L30-L32
to ensure it's used somewhere

Reply:
done